### ISSUE TITLE
wn::memory::aligned_realloc is broken on non-windows systems

### ISSUE THREAD

**Issue by devpow112** [WNProject/WNFramework#1]
The current implementation has a buffer overrun condition that causes a SEG_FAULT

```c++
wn_void* ptr = aligned_malloc(_new_size, _alignment);
ptr = memcpy(ptr, _ptr, _new_size); // buffer overrun causing segfault
aligned_free(_ptr);
return(ptr);
```
In order to fix this we must reimplement the alignment function on non-windows to store the alignmnet in the memory blob allocated or to do a normal realloc and then if the alignment is off align_malloc a block of the same size and copy it over.

**file:** [Libraries/WNMemory/inc/WNAllocation.h](https://github.com/WNProject/WNFramework/blob/e7e30138ed1822e0441c08bbcfa4a05340308006/Libraries/WNMemory/inc/WNAllocation.h)
**line:** 93

**Comment by AWoloszyn** [comment]
Fixed in 34da94f85f42810954212cacd473ad4efe582dae.

**Comment by AWoloszyn** [comment]
Fixed in 34da94f85f42810954212cacd473ad4efe582dae.
